src/12-02.zig

```diff
@@ -0,0 +1,104 @@
+const std = @import("std");
+const Allocator = std.mem.Allocator;
+
+fn LinkedList(comptime T: type) type {
+    return struct {
+        const Node = struct {
+            data: T,
+            next: ?*Node = null,
+        };
+
+        root: ?*Node = null,
+
+        allocator: Allocator,
+        const Self = @This();
+
+        fn init(allocator: Allocator) Self {
+            return Self{ .allocator = allocator };
+        }
+
+        fn add(self: *Self, value: T) !void {
+            var node = try self.allocator.create(Node);
+            node.* = Node{ .data = value };
+            if (self.root == null) {
+                self.root = node;
+                return;
+            }
+
+            var head: ?*Node = self.root;
+            while (head) |h| {
+                if (h.next == null) {
+                    h.next = node;
+                    return;
+                }
+
+                head = h.next;
+            }
+        }
+
+        fn search(self: *Self, value: T) bool {
+            var head: ?*Node = self.root;
+            var found: bool = false;
+            while (head) |h| {
+                if (h.data == value) {
+                    found = true;
+                    break;
+                }
+
+                head = h.next;
+            }
+            return found;
+        }
+
+        fn print(self: *Self, comptime printFunc: fn (object: T) void) void {
+            var node = self.root;
```

Comment:
```suggestion
            var head = self.root;
```

Keep variable name same with other functions.